aptos_sdk/async_client.py

```diff
@@ -568,25 +568,46 @@ async def transaction_pending(self, txn_hash: str) -> bool:
             raise ApiError(response.text, response.status_code)
         return response.json()["type"] == "pending_transaction"
 
-    async def wait_for_transaction(self, txn_hash: str) -> None:
-        """
-        Waits up to the duration specified in client_config for a transaction to move past pending
-        state.
-        """
-
-        count = 0
-        while await self.transaction_pending(txn_hash):
-            assert (
-                count < self.client_config.transaction_wait_in_seconds
-            ), f"transaction {txn_hash} timed out"
+    async def _wait_for_txn(self, txn_hash: str) -> None:
+        response = await self._get(endpoint=f"transactions/by_hash/{txn_hash}")
+        if response.status_code >= 400:
+            raise ApiError(response.text, response.status_code)
+        is_pending = response.json()["type"] == "pending_transaction"
+
+        if is_pending:
+            response = await self._get(endpoint=f"transactions/wait_by_hash/{txn_hash}")
+            if response.status_code >= 400:
+                raise ApiError(response.text, response.status_code)
+            is_pending = response.json()["type"] == "pending_transaction"
+
+        # retry if the txn is still pending
+        while is_pending:
+            response = await self._get(endpoint=f"transactions/by_hash/{txn_hash}")
```

Comment:
feels like we should just merge these....

```
while is_pending:
    response = wait_by_hash
    ...
```

no need for the break, asyncio.sleep, or count

Reply:
updated